src/templates/plugin.main.js

```diff
@@ -149,7 +151,13 @@ export default async (context) => {
 
     // Handle cookie option to prevent multiple redirections
     if (finalLocale && (!useCookie || alwaysRedirect || !app.i18n.getLocaleCookie())) {
-      if (finalLocale !== app.i18n.locale) {
+      if (strategy !== STRATEGIES.NO_PREFIX && route.path !== '/') {
+        if (useCookie) {
+          app.i18n.setLocaleCookie(finalLocale)
+        } else {
+          return finalLocale
+        }
```

Comment:
It should be possible to handle it better by not saving the cookie on almost every navigation and also this function should return false if the locale is not supposed to change.